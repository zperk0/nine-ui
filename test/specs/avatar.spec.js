import NAvatar from '../../packages/n-avatar';
import { mount } from '@vue/test-utils';

describe('NAvatar', () => {
  it('renders correctly', () => {
    const wrapper = mount({
      render () {
        return <NAvatar>Follow</NAvatar>;
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('render img when has src', () => {
    const wrapper = mount({
      render () {
        return <NAvatar src="https://avatars2.githubusercontent.com/u/12592949?s=460&v=4" />;
      }
    });
    expect(wrapper.find('.navatar-img').exists()).toBe(true);
  });
  it('render shape correctly', () => {
    const wrapper = mount({
      render () {
        return <NAvatar shape="circle" />;
      }
    });
    expect(wrapper.find('.navatar-circle').exists()).toBe(true);
  });
  it('render size correctly', () => {
    const wrapper = mount({
      render () {
        return <NAvatar size="lg" />;
      }
    });
    expect(wrapper.find('.navatar-lg').exists()).toBe(true);
  });
});
