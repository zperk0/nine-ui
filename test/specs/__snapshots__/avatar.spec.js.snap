// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NAvatar renders correctly 1`] = `
<div class="navatar navatar-md navatar-square" style="background: rgb(204, 204, 204);">
  <div>Follow</div>
</div>
`;
